src/lib/running.ml

```diff
@@ -32,14 +37,14 @@ let init ?(size=1) o = { size
              ; var    = 0.0
              }
 
-let update ?(size=1) t v =
+let update ?(size=1) ?(mean_update=default_mean_update) t v =
   if t.size = 0
   then init ~size v
   else let size_f = float size in
        let n_sum = t.sum +. size_f *. v in
        let n_sum_sq = t.sum_sq +. size_f *. v *. v in
```

Comment:
I thought the same thing intuitively but my test doesn't pass if I change
it to that. If you think of it as N independent contributions of v^2 I feel
like the interpretation is sound.
On May 6, 2015 7:41 PM, "Leonid Rozenberg" <notifications@github.com> wrote:

> In src/lib/running.ml
> <https://github.com/rleonid/oml/pull/36#discussion_r29813668>:
>
> > @@ -32,14 +37,14 @@ let init ?(size=1) o = { size
> >               ; var    = 0.0
> >               }
> >
> > -let update ?(size=1) t v =
> > +let update ?(size=1) ?(mean_update=default_mean_update) t v =
> >    if t.size = 0
> >    then init ~size v
> >    else let size_f = float size in
> >         let n_sum = t.sum +. size_f *. v in
> >         let n_sum_sq = t.sum_sq +. size_f *. v *. v in
>
> @struktured <https://github.com/struktured> shouldn't this be?
>
> let n_sum_sq = t.sum_sq +. size_f *. size_f *. v *. v in
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rleonid/oml/pull/36/files#r29813668>.
>